parser/src/dataset/mod.rs

```diff
@@ -420,14 +421,15 @@ pub trait IntoTokens {
     /// The iterator type through which tokens are obtained.
     type Iter: Iterator<Item = DataToken>;
 
-    /// Convert the value into tokens.
-    fn into_tokens(self) -> Self::Iter;
+    /// Convert the value into tokens. The parameter value determines if the sequence length in bytes
+    /// might be invalid and should either be recalculated or marked as undefined.
+    fn into_tokens(self, invalidate_sq_length: bool) -> Self::Iter;
```

Comment:
Ah yes, I see. That is indeed a breaking change. Might be a bit safer to have that struct indeed. Ideally we'd not have this at all of course.. It feels a bit that converting into tokens should be unambiguous and not require any parameters.

That being  said, that would require computing the adequate lengths during token generation, which might be very difficult to do. Or we could do something similar to what we do for strings and ignore the lengths. But that would require writing nested data into intermediate buffers to check the lengths. It might not be that bad though, since we can skip it for pixel data, and the rest should be smallish?

But having said all that, maybe the `IntoTokenOptions` enum is the way to go now.

Reply:
One thing I was thinking about: if we go for options, what should the default behavior be? Easiest is perhaps to keep things as they are, but it is a bit unsafe as it could potentially lead to invalid DICOM (albeit probably only if the charset changes). But a case could be made to make the default to invalidate all lengths, and only respect sequence / item lengths if we are sure they are safe. But that would require all downstream `IntoToken` implementations to change too.

Anyways, your call, just something I was thinking about.

Reply:
If possible, we should purport this option being enabled as "definitely invalidate sequence lengths because we _definitely_ cannot rely on the lengths at this point", whereas if the option is disabled, we try to retain lengths within the implementation's ability. These guarantees should hopefully be enough to attain consistency in nearly all use cases.

Reply:
Ok, let's go for that then :+1: I'll make the change.